app/controllers/samples_controller.rb

```diff
@@ -802,33 +802,37 @@ def report_v2
     pipeline_run = select_pipeline_run(@sample, params[:pipeline_version])
     background_id = get_background_id(@sample, params[:background])
 
-    # Don't cache the response until the pipeline run is report-ready
-    # so the displayed pipeline run status will be updated correctly.
-    skip_cache = !pipeline_run.report_ready? || params[:skip_cache] || false
-
-    report_info_params = pipeline_run.report_info_params
-    # If the pipeline_version wasn't passed in from the client-side,
-    # then set it to version for the selected default pipeline run.
-    if params[:pipeline_verison].nil?
-      params[:pipeline_version] = pipeline_run.pipeline_version
-    end
-    cache_key = PipelineReportService.report_info_cache_key(
-      request.path,
-      report_info_params
-        .merge(
-          params
-            .reject { |_, v| v.blank? }
-            .permit(report_info_params.keys)
-        ).merge(
-          background_id: background_id
-        ).symbolize_keys
-    )
-    httpdate = Time.at(report_info_params[:report_ts]).utc.httpdate
-
-    json =
-      fetch_from_or_store_in_cache(skip_cache, cache_key, httpdate, "PipelineReport") do
-        PipelineReportService.call(pipeline_run, background_id)
+    if pipeline_run
+      # Don't cache the response until the pipeline run is report-ready
+      # so the displayed pipeline run status will be updated correctly.
+      skip_cache = !pipeline_run.report_ready? || params[:skip_cache] || false
+
+      report_info_params = pipeline_run.report_info_params
+      # If the pipeline_version wasn't passed in from the client-side,
+      # then set it to version for the selected default pipeline run.
+      if params[:pipeline_verison].nil?
+        params[:pipeline_version] = pipeline_run.pipeline_version
       end
+      cache_key = PipelineReportService.report_info_cache_key(
+        request.path,
+        report_info_params
+          .merge(
+            params
+              .reject { |_, v| v.blank? }
+              .permit(report_info_params.keys)
+          ).merge(
+            background_id: background_id
+          ).symbolize_keys
+      )
+      httpdate = Time.at(report_info_params[:report_ts]).utc.httpdate
+
+      json =
+        fetch_from_or_store_in_cache(skip_cache, cache_key, httpdate, "PipelineReport") do
+          PipelineReportService.call(pipeline_run, background_id)
+        end
+    else
```

Comment:
We call this with pipeline run nil?

Reply:
I guess there is a particular status to it...

Reply:
Yup, the call has a check to see if the pipeline run is nil and fills out the status as "waiting."